zeppelin-web/src/app/interpreter/interpreter.controller.js

```diff
@@ -22,6 +22,171 @@ angular.module('zeppelinWebApp').controller('InterpreterCtrl',
     $scope.showRepositoryInfo = false;
     $scope._ = _;
 
+    // auto complete related
+    $scope.suggestions = [];
+    $scope.selectIndex = -1;
+    var selectedUser = '';
+    var selectedUserIndex = 0;
+    var previousSelectedList = [];
+    var searchText = [];
+
+    function updatePreviousList() {
```

Comment:
`updatePreviousList` defined multiple times; for loop inside (or even the whole function) can be replaced with [_.clone](https://lodash.com/docs#clone)